core/oa_hash_map.h

```diff
@@ -0,0 +1,571 @@
+/*************************************************************************/
+/*  oa_hash_map.h                                                        */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2017 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2017 Godot Engine contributors (cf. AUTHORS.md)    */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+#ifndef OA_HASH_MAP_H
+#define OA_HASH_MAP_H
+
+#include "hashfuncs.h"
+#include "math_funcs.h"
+#include "os/copymem.h"
+#include "os/memory.h"
+
+#define OA_HASH_MAP_INITIAL_LOCAL_STORAGE
```

Comment:
Why is this just statically defined here, if it's not going to be disabled by buildsystem just take it out?

Reply:
I asked on the IRC... I don't know where to put it :D (the compiler flag I mean)

Reply:
I guess the question was "why are there `#ifdef`s at all if this can't be disabled?".

Reply:
It can be disabled, I just didn't put this in a Scons file because I don't know where. core/SCsub doesn't seem right.

Reply:
Well the question is "do we want to disable it?". If not, I prefer we don't bloat the CCFLAGS with yet another super long define (remember that Windows commits seppuku when the compiler command is too long). And then we can just keep this static define here if some devs want to disable it manually for testing.

Reply:
My thought was that it should be enabled by default, just like you said, if somebody wants to disable it (because the object gets to "fat" or something) then he can do it here. 

I would've added it as a default flag, but realistically I don't see a reason why it should be disabled.

Reply:
Then it's good as is IMO.